src/main/java/com/google/api/generator/gapic/composer/utils/ClassNames.java

```diff
@@ -120,4 +122,9 @@ private static String monolithBackwardsCompatibleName(String rawServiceName) {
         ? rawServiceName.replace("IAM", "Iam")
         : rawServiceName;
   }
+
+  // For region tags, the service name should not include `Client` in the name
+  public static String getPureServiceName(String serviceClassName) {
+    return Iterables.getFirst(Splitter.on("Client").split(serviceClassName), serviceClassName);
```

Comment:
If you decide to get service name by splitting the String, I would suggest extract the `Client` here and [here](https://github.com/googleapis/gapic-generator-java/blob/main/src/main/java/com/google/api/generator/gapic/composer/utils/ClassNames.java#L27) to a constant, in case it changes in the future.

Reply:
Thanks - updated.